internal/services/hsm/cloud_hardware_security_module_cluster_resource.go

```diff
@@ -0,0 +1,466 @@
+	PrivateEndpointConnections []PrivateEndpointConnectionModel `tfschema:"private_endpoint_connections"`
+	StatusMessage              string                           `tfschema:"status_message"`
+}
+
+type CloudHsmPropertiesModel struct {
+	Fqdn         string `tfschema:"fqdn"`
+	State        string `tfschema:"state"`
+	StateMessage string `tfschema:"state_message"`
+}
+
+type PrivateEndpointConnectionModel struct {
+	Id                                string                                   `tfschema:"id"`
+	Name                              string                                   `tfschema:"name"`
+	Type                              string                                   `tfschema:"type"`
+	GroupIds                          []string                                 `tfschema:"group_ids"`
+	PrivateEndpoint                   []PrivateEndpointModel                   `tfschema:"private_endpoint"`
+	PrivateLinkServiceConnectionState []PrivateLinkServiceConnectionStateModel `tfschema:"private_link_service_connection_state"`
+}
+
+type PrivateEndpointModel struct {
+	Id string `tfschema:"id"`
+}
+
+type PrivateLinkServiceConnectionStateModel struct {
+	Status          string `tfschema:"status"`
+	Description     string `tfschema:"description"`
+	ActionsRequired string `tfschema:"actions_required"`
+}
+
+var _ sdk.ResourceWithUpdate = CloudHardwareSecurityModuleClusterResource{}
+
+type CloudHardwareSecurityModuleClusterResource struct{}
+
+func (CloudHardwareSecurityModuleClusterResource) ResourceType() string {
+	return "azurerm_cloud_hardware_security_module_cluster"
+}
+
+func (CloudHardwareSecurityModuleClusterResource) ModelObject() interface{} {
+	return &CloudHardwareSecurityModuleClusterModel{}
+}
+
+func (CloudHardwareSecurityModuleClusterResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return cloudhsmclusters.ValidateCloudHsmClusterID
+}
+
+func (r CloudHardwareSecurityModuleClusterResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ValidateCloudHsmClusterName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		// only user assigned identity is supported even though the swagger shows system assigned identity
+		"identity": commonschema.UserAssignedIdentityOptional(),
+
+		"auto_generated_domain_name_label_scope": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			Default:      string(cloudhsmclusters.AutoGeneratedDomainNameLabelScopeTenantReuse),
+			ValidateFunc: validation.StringInSlice(cloudhsmclusters.PossibleValuesForAutoGeneratedDomainNameLabelScope(), false),
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r CloudHardwareSecurityModuleClusterResource) Attributes() map[string]*pluginsdk.Schema {
```

Comment:
is it necessary to expose all of these fields? For example, in what scenarios will `activation_state`, `hsms.state`, `state.state_message`, `private_link_service_connection_state`, `status_message` be used?